### PR TITLE
Enable Basic Auth file user store support

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
This PR enables the Basic Auth file user store support which was disabled by https://github.com/ballerina-platform/module-ballerina-auth/pull/60 due to a limitation of `configurable` feature.

Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/862

Related to https://github.com/ballerina-platform/ballerina-standard-library/issues/584

### CODE REVIEW

File: auth-ballerina/auth_constants.bal [hunk 1]
```diff
@@ -20,6 +20,9 @@ public const string DEFAULT_CHARSET = "UTF-8";
 # Prefix used to denote special configuration values.
 public const string CONFIG_PREFIX = "@";
 
+# Prefix used to denote that the config value is a encrypted.
```
Comment (chamil321) [inline auth-ballerina/auth_constants.bal:23]:
```suggestion
# Prefix used to denote that the config value is encrypted.
```

File: auth-ballerina/listener_file_user_store_basic_auth_provider.bal [hunk 2]
```diff
@@ -1,124 +1,109 @@
...
+# scopes = ["scope1", "scope2"]
+# ```
+public class ListenerFileUserStoreBasicAuthProvider {
+
+    *ListenerBasicAuthProvider;
+
+    # Provides authentication based on the provided configurations.
+    #
+    # + fileUserStoreConfig - file user store configurations
+    public isolated function init(FileUserStoreConfig fileUserStoreConfig = {}) {
```
Comment (chamil321) [inline auth-ballerina/listener_file_user_store_basic_auth_provider.bal:54]: Do we need this init function? Seems like the `fileUserStoreConfig` has not been used inside the method

File: auth-ballerina/listener_ldap_user_store_basic_auth_provider.bal [hunk 3]
```diff
@@ -129,22 +129,23 @@ public class ListenerLdapUserStoreBasicAuthProvider {
         [string, string] [username, password] = check extractUsernameAndPassword(credential);
         Error? authenticated = authenticateWithLdap(self.ldapConnection, username, password);
         if (authenticated is Error) {
-            return prepareError("Failed to authenticate LDAP user store with username: " + username, authenticated);
-        }
-        string[]|Error groups = getLdapGroups(self.ldapConnection, username);
-        if (groups is Error) {
-            return prepareError("Failed to get groups from LDAP user store with the username: " + username, groups);
+            return prepareError("Failed to authenticate username: " + username + " with LDAP user store.", authenticated);
```
Comment (chamil321) [inline auth-ballerina/listener_ldap_user_store_basic_auth_provider.bal:132]:
```suggestion
            return prepareError("Failed to authenticate username: '" + username + "' with LDAP user store.", authenticated);
```

File: auth-ballerina/listener_ldap_user_store_basic_auth_provider.bal [hunk 4]
```diff
@@ -129,22 +129,23 @@ public class ListenerLdapUserStoreBasicAuthProvider {
...
         UserDetails userDetails = {
-            username: username,
-            scopes: checkpanic groups
+            username: username
         };
+        string[]|Error? groups = getLdapGroups(self.ldapConnection, username);
+        if (groups is string[]) {
+            userDetails.scopes = groups;
+        } else if (groups is Error) {
+            return prepareError("Failed to get groups for the username: " + username + " from LDAP user store.", groups);
```
Comment (chamil321) [inline auth-ballerina/listener_ldap_user_store_basic_auth_provider.bal:141]:
```suggestion
            return prepareError("Failed to get groups for the username: '" + username + "' from LDAP user store.", groups);
```

File: auth-ballerina/listener_file_user_store_basic_auth_provider.bal [hunk 5]
```diff
@@ -1,124 +1,109 @@
...
+                UserDetails userDetails = {
+                    username: username
+                };
+                string[]? scopes = authInfo?.scopes;
+                if (scopes is string[]) {
+                    userDetails.scopes = scopes;
+                }
+                return userDetails;
+            }
+            return prepareError("Failed to authenticate username: " + username + " from file user store.");
```
Comment (chamil321) [inline auth-ballerina/listener_file_user_store_basic_auth_provider.bal:83]:
```suggestion
            return prepareError("Failed to authenticate username: '" + username + "' from file user store.");
```

File: auth-ballerina/listener_file_user_store_basic_auth_provider.bal (hunk 2 above)
Comment (ldclakmal) [reply auth-ballerina/listener_file_user_store_basic_auth_provider.bal:54]: Yes. This was kept blank intentionally for future usages. In order to support improvements without braking changes. Also, the usage of this API from HTTP module (declarative auth design) will be problematic without this init function. Will add a comment inside the function.

**Review by chamil321 — APPROVED** [review]

**Pull request merged**